aws_s3_bucket.tf

```diff
@@ -61,31 +73,47 @@ resource "aws_s3_bucket" "s3VersioningMfaFalse" {
       }
     }
   }
-  
+
   versioning {
-    enabled = false
-    mfa_delete = false
+    enabled    = true
+    mfa_delete = true
   }
 }
 
 resource "aws_s3_bucket" "allUsersReadAccess" {
   bucket = "my-tf-test-bucket"
-  acl    = "public-read"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
 }
 
 resource "aws_s3_bucket" "authUsersReadAccess" {
   bucket = "my-tf-test-bucket"
-  acl    = "authenticated-read"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
 }
 
 resource "aws_s3_bucket" "allUsersWriteAccess" {
   bucket = "my-tf-test-bucket"
-  acl    = "public-read-write"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
 }
 
 resource "aws_s3_bucket" "allUsersReadWriteAccess" {
   bucket = "my-tf-test-bucket"
-  acl    = "public-read-write"
+  acl    = "private"
+
+  versioning {
+    enabled = true
```

Comment:
**Accurics remediation**
 
**[HIGH]: S3 bucket accessible to all authenticated users.**
 
**Vulnerability**
S3 bucket Access is allowed to all AWS Account Users.
 
**Remediation**
S3 bucket Access should be limited to specific users and roles.

source: accurics-